### PR TITLE
feat: adding workload version

### PR DESCRIPTION
**Pull request by markbeierl**
# Description

Checks the running container for a specific /etc/workload-version file. If it exists, the content of that file is reported in the app version field.

# Checklist:

- [X] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that validate the behaviour of the software
- [X] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -110,6 +111,7 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):
             event.add_status(WaitingStatus("Waiting for container to be ready"))
             logger.info("Waiting for container to be ready")
             return
+        self.unit.set_workload_version(self._get_workload_version())
```
Comment (gruyaume) [inline src/charm.py:114]: Why do we place this inside of collect unit status? It feels like we are adding a side effect to a method that had 1 purpose

Comment (markbeierl) [reply src/charm.py:114]: It is part of the status of the unit. Where else would it make sense to put it?

Comment (markbeierl) [reply src/charm.py:114]: The only other place is in configure, but we are not configuring the workload, we are reporting info about the charm. Would that be a better location?

Comment (gruyaume) [reply src/charm.py:114]: Yeah I agree that configure is probably not the right place either. Ok let's keep it here, maybe let's update the docstring to reflect that this hook also sets the workload version.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**